docs/concepts/macros/sqlmesh_macros.md

```diff
@@ -911,6 +911,37 @@ The methods are available because the `column` argument is parsed as a SQLGlot [
 
 Column expressions are sub-classes of the [Condition class](https://sqlglot.com/sqlglot/expressions.html#Condition), so they have builder methods like [`between`](https://sqlglot.com/sqlglot/expressions.html#Condition.between) and [`like`](https://sqlglot.com/sqlglot/expressions.html#Condition.like).
 
+#### Accessing model schemas
+
+Model schemas can be accessed within a Python macro function through its evaluation context's `column_to_types` method, when they can be statically determined.
+
+For example, consider the following macro function which aims to rename the columns of a target model by adding a prefix to them:
+
+```python linenums="1"
+from sqlglot import exp
+from sqlmesh.core.macros import macro
+
+@macro()
+def foo(evaluator, model_name, prefix):
+    prefix = prefix.name
+    renamed_projections = []
+
+    for name, dtype in evaluator.columns_to_types(model_name.sql()).items():
+        new_name = prefix + name
+        renamed_projections.append(exp.cast(exp.column(name), dtype).as_(new_name))
+
+    return renamed_projections
+```
+
+Having access to the schema of an upstream model can be useful for various reasons:
```

Comment:
Is it fair to say that the most common use case will be when the columns in upstream tables are either unknown or dynamic?

Reply:
I don't think so: unknown columns can't be returned by `columns_to_types` because that'd mean we can infer them statically and they wouldn't really be "unknown". Unless I misread your comment?

Correct me if I'm wrong, but my understanding is that the most common use case is creating a dependency inversion layer between e.g. external models and their downstream consumers by applying certain transformations (renames, casts, etc - basically what I mentioned in the bullets)